modin/pandas/utils.py

```diff
@@ -263,6 +263,85 @@ def to_pandas(df):
     return pandas_df
 
 
+# Indexing
+#  Generate View Copy Helpers
+#  Dependency:
+#      _generate_block takes in argument generated by repartition_coord_df
+#      _generate_block calls extract_block
+#  extract_block was moved to the end of the file
+
+
+def _repartition_coord_df(old_coord_df, npartition):
+    """Repartition the (view of) coord_df by npartition
+
+    This function is used when old_coord_df is not contigous.
+    For example, it turns:
+
+        partition index_within_partition
+    i0  0         0
+    i6  3         2
+
+    into
+
+        partition index_within_partition
+    i0  0         0
+    i6  0         1
+
+    Note(simon):
+        The resulting npartition will be <= npartition
+        passed in.
+    """
+    length = len(old_coord_df)
+    chunksize = (len(old_coord_df) // npartition) + 1
+
+    # genereate array([0, 0, 0, 1, 1, 1, 2])
+    partitions = np.repeat(np.arange(npartition), chunksize)[:length]
+
+    # generate array([0, 1, 2, 0, 1, 2, 0])
+    final_n_partition = np.max(partitions)
+    idx_in_part = np.tile(np.arange(chunksize), final_n_partition + 1)[:length]
+
+    final_df = pandas.DataFrame(
+        {
+            'partition': partitions,
+            'index_within_partition': idx_in_part
+        },
+        index=old_coord_df.index)
+
+    return final_df
+
+
+def _generate_blocks(old_row, new_row, old_col, new_col,
+                     block_partition_2d_oid_arr):
+    """
+    Given {old, new} x {row, col} meta data and the old 2d oid array,
+    return a new 2d oid array.
+    """
+
+    def join(old, new):
+        return new.merge(
+            old, left_index=True, right_index=True, suffixes=('_new', '_old'))
+
+    row_grouped = join(old_row, new_row).groupby('partition_new')
+    col_grouped = join(old_col, new_col).groupby('partition_new')
+
+    oid_lst = []
+    for row_idx, row_lookup in row_grouped:
+        for col_idx, col_lookup in col_grouped:
+            oid = extract_block.remote(
+                block_partition_2d_oid_arr,
+                row_lookup,
+                col_lookup,
+                col_name_suffix='_old')
+            oid_lst.append(oid)
+    return np.array(oid_lst).reshape(len(row_grouped), len(col_grouped))
+
+
+# Indexing
```

Comment:
What are these used for? We haven't used this convention elsewhere, do you think we should?

Reply:
We are going to refractor utils into a package, and these functions will be in separate files named under `indexing`, and this comment block will be part of the file level description.

Reply:
Sounds good. I like the idea of using these and have adopted the general concept in the rewrite.